File: python/code/78.py
class Solution:
    def subsets(self, nums):
        res = [[]]
        for i in nums:
            res = res + [[i] + num for num in res]
        return res

        ret = []
        for i in range(2 ** len(nums)):
            element = []
            cnt = 0
            while i > 0:
                if i & 1 == 1:
                    element.append(nums[cnt])
                cnt += 1
                i >>= 1
            ret.append(element)
        return ret

File: C++/code/78.cpp
class Solution {
public:
    vector<vector<int>> ans;
    vector<int> t;
    
    void dfs(int cur, vector<int>& nums) {
        if (cur == nums.size()) {
            ans.push_back(t);
            return;
        }
        t.push_back(nums[cur]);
        dfs(cur + 1, nums);
        t.pop_back();
        dfs(cur + 1, nums);
    }
    
    vector<vector<int>> subsets(vector<int>& nums) {
        dfs(0, nums);
        return ans;
    }
};